src/Microsoft.AspNetCore.Server.Kestrel.Core/KestrelServer.cs

```diff
@@ -208,6 +208,19 @@ private void ValidateOptions()
                 throw new InvalidOperationException(
                     CoreStrings.FormatMaxRequestBufferSmallerThanRequestHeaderBuffer(Options.Limits.MaxRequestBufferSize.Value, Options.Limits.MaxRequestHeadersTotalSize));
             }
+
+            if (Options.Limits.DefaultRequestBodyExtendedTimeout.HasValue &&
+                Options.Limits.DefaultRequestBodyExtendedTimeout <= Options.Limits.DefaultRequestBodyTimeout)
+            {
+                throw new InvalidOperationException(
+                    CoreStrings.FormatRequestBodyExtendedTimeoutSmallerThanTimeout(Options.Limits.DefaultRequestBodyExtendedTimeout, Options.Limits.DefaultRequestBodyTimeout));
+            }
+
+            if (Options.Limits.DefaultRequestBodyExtendedTimeout.HasValue ^ Options.Limits.DefaultRequestBodyMinimumDataRate.HasValue)
```

Comment:
Since they are both required to be null or set, can we make this enforceable as a compile-time check? Makes the API slightly easier to use

Ideas:
```c#
// add a type
public class KestrelServerLimits
{
    public DataRateLimit DefaultRequestBodyRate { get; set; }
}

public class DataRateLimit
{
    public TimeSpan Timeout { get;set;}
    public double Rate { get; set; }
}

```
```c#
// add a tuple literal
public class KestrelServerLimits
{
    public (TimeSpan timespan, double rate) DefaultRequestBodyRate { get; set; }
}
```
```c#
// add a tuple type
public class KestrelServerLimits
{
    public Tuple<TimeSpan, double> DefaultRequestBodyRate { get; set; }
}
```